components/markdown_image/markdown_image.jsx

```diff
@@ -66,17 +71,25 @@ export default class MarkdownImage extends React.PureComponent {
         return parseInt(height, 10);
     }
 
-    showModal = (e) => {
-        if (!this.props.imageIsLink) {
+    showModal = (e, link, imageMetadata, extension, name) => {
+        if (!this.props.imageIsLink && extension) {
```

Comment:
should we add extensions to the check too, i added since it was in https://github.com/mattermost/mattermost-webapp/pull/9334/files#diff-ececdf452d72bd2b2ad7c33364a8648ae2e29476f3f0be17a1787cdb98c1c532L185

Reply:
Makes sense to me. I'm not sure what the failure case looks like for when there isn't an extension though, since with this it would just not pop a modal vs. before where it would pop a modal but not load an image.